source/common/http/header_map_impl.cc

```diff
@@ -443,57 +434,79 @@ void HeaderMapImpl::addCopy(const LowerCaseString& key, uint64_t value) {
     return;
   }
   HeaderString new_key;
-  new_key.setCopy(key.get().c_str(), key.get().size());
+  new_key.setCopy(key.get());
   HeaderString new_value;
   new_value.setInteger(value);
   insertByKey(std::move(new_key), std::move(new_value));
   ASSERT(new_key.empty());   // NOLINT(bugprone-use-after-move)
   ASSERT(new_value.empty()); // NOLINT(bugprone-use-after-move)
 }
 
-void HeaderMapImpl::addCopy(const LowerCaseString& key, const std::string& value) {
+void HeaderMapImpl::addCopy(const LowerCaseString& key, absl::string_view value) {
   auto* entry = getExistingInline(key.get());
   if (entry != nullptr) {
     const uint64_t added_size = appendToHeader(entry->value(), value);
     addSize(added_size);
     return;
   }
   HeaderString new_key;
-  new_key.setCopy(key.get().c_str(), key.get().size());
+  new_key.setCopy(key.get());
   HeaderString new_value;
-  new_value.setCopy(value.c_str(), value.size());
+  new_value.setCopy(value);
   insertByKey(std::move(new_key), std::move(new_value));
   ASSERT(new_key.empty());   // NOLINT(bugprone-use-after-move)
   ASSERT(new_value.empty()); // NOLINT(bugprone-use-after-move)
 }
 
-void HeaderMapImpl::setReference(const LowerCaseString& key, const std::string& value) {
+void HeaderMapImpl::append(const LowerCaseString& key, absl::string_view value) {
+  bool found = false;
+  for (HeaderEntryImpl& header : headers_) {
```

Comment:
Sorry just saw this, but independently, does this work if these methods are called for inline headers? I think you need to call `getExistingInline` first like other similar methods do? Same below for the other new method?

Reply:
Hmmm... Nope they work fine. `headers_` contains the inline headers as well. The other methods call `getExistingInline` because their behavior differs for both cases as well. I added tests for this though (calling `append` and `setCopy` on inline headers without the direct inline accessors).

Reply:
Right, sorry, sounds good.